### PR TITLE
Add throwable capture balls

### PR DESCRIPTION
**Pull request by keyz182**
- Throw to capture
  - if non-friendly and downed, pawn is captured as prisoner
  - if non-friendly and not downed, nothing happens
  - if friendly, pawn is stored normally
-  Throw to release
   - Releases the pawn at the position hit

### CODE REVIEW

**Comment by keyz182** [comment]
Implemented separately from the crystal for now - if you want I can merge them into one.

**Review by feldoh — CHANGES_REQUESTED** [review]

File: 1.5/Defs/ThingDefs_Items/Items_Exotic.xml [hunk 1]
```diff
@@ -152,6 +161,133 @@
...
+                <maxStoredPawns>1</maxStoredPawns>
+                <releaseOption>true</releaseOption>
+                <convertOption>true</convertOption>
+                <appendOfName>true</appendOfName>
+                <lightEffect>true</lightEffect>
+                <releaseEffect>Skip_Entry</releaseEffect>
+                <storeEffect>Skip_Entry</storeEffect>
+                <charges>1</charges>
+                <useCharges>true</useCharges>
+                <destroyOnZeroCharges>true</destroyOnZeroCharges>
```
[the remark this answers is not in the document]
Comment (keyz182) [reply 1.5/Defs/ThingDefs_Items/Items_Exotic.xml:252]: Yup, that's right. That said, I don't think this is having any effect anymore, as I'm not manually destoying the ball from the projectile on capture start. Might need to revisit that behaviour as it essentially lets you keep throwing until you capture.

File: 1.5/Source/PawnStorages/PawnStorages/CompPawnStorage.cs [hunk 2]
```diff
@@ -237,14 +244,28 @@ public virtual void ApplyNeedsForStoredPeriodFor(Pawn pawn)
         }
     }
 
+    public void TransferPawn(CompPawnStorage otherStore, Pawn pawn)
```
[the remark this answers is not in the document]
Comment (keyz182) [reply 1.5/Source/PawnStorages/PawnStorages/CompPawnStorage.cs:255]: What do you mean by base comp here? I thought CompPawnStorage was the base comp.

File: 1.5/Source/PawnStorages/PawnStorages/CompPawnStorage.cs [hunk 3]
```diff
@@ -111,7 +111,14 @@ public override string TransformLabel(string label)
         }
         else
         {
-            transformLabelCache = $"{base.TransformLabel(label)} {"PS_Filled".Translate()}";
+            if (Props.MaxStoredPawns == 1)
```
[the remark this answers is not in the document]
Comment (keyz182) [reply 1.5/Source/PawnStorages/PawnStorages/CompPawnStorage.cs:114]: Might have missed something, but I thought all the label code was here in this function?

[the remark this answers is not in the document]
Comment (feldoh) [reply 1.5/Source/PawnStorages/PawnStorages/CompPawnStorage.cs:114]: 🤦 I thought this was a special one for absolutely no reason

File: 1.5/Source/PawnStorages/PawnStorages/CompPawnStorage.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (feldoh) [reply 1.5/Source/PawnStorages/PawnStorages/CompPawnStorage.cs:255]: 🤦 I thought this was a special one for absolutely no reason

**Review by feldoh — APPROVED** [review]

**Pull request merged**